### PR TITLE
Set up Playwright screenshot testing

### LINKED ISSUE #557: Save screenshots as side-effect in Playwright tests
### What's your idea?

To help keep track of all visual changes made to the website from now on, I propose that we use Playwright to take screenshots of each page configured to do so. In this proposal, I am considering taking the screenshots as a side-effect of the Playwright tests, since they do the job of spinning up each page.

Related: #336, #120

### PR DESCRIPTION
**Pull request by EthanThatOneKid**
### What we can do now

`npm t` has now been upgraded to include parallelized Playwright tests and screenshot diff testing. To specifically test the Playwright tests and not the Vitest tests, you may run `npm run test:browser`. Make some changes on screenshot-tested pages and give it a try!

```bash
npm t
```

### Change log
- Split up `/tests/test.ts` tests into each page of the `/src/routes/` directory.
- Added `projectsFromMatrix` helper to generate a matrix of screenshots to be taken per page.
- Added `vitest.config.ts` to exclude Playwright tests from the Vitest testing.
- Generated the first official set of screenshots in `/src/routes/screenshots/`.
- This PR goes on to add the screenshot testing boilerplate for each page endpoint of the website.
  - [x] Added screenshot tests for 404 error page.
  - [x] Added screenshot tests for `/1st` page.
  - [x] Added screenshot tests for `/about` page.
  - [x] Added screenshot tests for `/blog` page.
  - [x] Added screenshot tests for `/blog/:id` page.
  - [x] Added screenshot tests for `/events` page.
  - [x] Added screenshot tests for `/nodebuds` page.
  - [x] Added screenshot tests for `/privacy` page.
  - [x] Added screenshot tests for `/quiz` page.
  - [x] Added screenshot tests for `/teams` page.

Resolves #557.

### CODE REVIEW

**Comment by anguzz** [comment]
HUGE 🧠

File: package.json [hunk 1]
```diff
@@ -19,15 +19,15 @@
...
+    "test:browser": "playwright test",
     "check": "svelte-kit sync && svelte-check --tsconfig ./tsconfig.json --compiler-warnings \"css-unused-selector:ignore,unused-export-let:error\"",
     "lint": "eslint . --fix",
     "format": "prettier --write .",
     "host": "npm run all && npm run preview",
     "all": "npm run lint && npm run format && npm run check && npm run build && npm t"
   },
   "devDependencies": {
-    "@playwright/test": "^1.25.0",
+    "@playwright/test": "^1.28.0-alpha-oct-12-2022",
```
Comment (charliettaylor) [inline package.json]: Is there a specific reason for using this version? Using alpha seems unnecessary

Comment (EthanThatOneKid) [reply package.json]: > Is there a specific reason for using this version? Using alpha seems unnecessary

Good catch. While testing, I updated to the `@next` version, but you are right that’s probably not necessary. I’ll set it to the `@latest` version instead.

Comment (EthanThatOneKid) [reply package.json]: Resolved in <https://github.com/EthanThatOneKid/acmcsuf.com/commit/829a50bfa2b0cdad02a5d70ecfb0814d7e20da26>.

**Comment by EthanThatOneKid** [comment]
This PR is intended to introduce a new feature as a trial run. If the _new feature_ (Playwright screenshot testing) works out to be more of a hinderance than a convenience, then I will remove this feature sometime near the end of the Fall 2022 semester.

**Review by k4rni — APPROVED** [review]
LGTM

**Review by jaasonw — APPROVED** [review]
i did not look at it so therefore i see no issues with it

**Review by charliettaylor — APPROVED** [review]
image

**Pull request merged**